lib/location/method_gnss.c

```diff
@@ -158,23 +213,87 @@ int method_gnss_location_get(const struct loc_method_config *config)
 		use_case = NRF_MODEM_GNSS_USE_CASE_MULTIPLE_HOT_START;
 		err |= nrf_modem_gnss_use_case_set(use_case);
 		if (!err) {
-			fix_attemps_remaining = gnss_config->num_consecutive_fixes;
+			fix_attemps_remaining = gnss_config.num_consecutive_fixes;
 		}
 		break;
 	}
 
 	if (err) {
 		LOG_ERR("Failed to configure GNSS");
-		return -EINVAL;
+		loc_core_event_cb_error();
+		running = false;
 	}
 
 	err = nrf_modem_gnss_start();
 	if (err) {
 		LOG_ERR("Failed to start GNSS");
-		return err;
+		loc_core_event_cb_error();
+		running = false;
 	}
 
 	LOG_DBG("GNSS started");
+}
 
-	return err;
+void method_gnss_modem_sleep_notif_subscribe(uint32_t threshold_ms)
```

Comment:
Since we don't have de-init function for location API, there is no place to unsubscribe from notifications.

Reply:
I have another comment about `de-init` too. We can think about it later but not sure yet that we need such.
However, I'm also thinking if we should subscribe to sleep notifications in `init()` or whether we should do it in `method_gnss_location_get()`. This is just something we should think about.

Reply:
I think that there isn't way to query/poll PSM state, right? and that mandates us to keep us updated started from bootup of the state and also between the gnss usages.

Reply:
Ok, in that case we are unaware of PSM state potentially when the location_request comes in.

Reply:
Closing this. We are making a timeout into the PSM wait so it won't get stuck.